types/agent-base/index.d.ts

```diff
@@ -0,0 +1,44 @@
+// Type definitions for agent-base 4.2
+// Project: https://github.com/TooTallNate/node-agent-base#readme
+// Definitions by: Christopher Quadflieg <https://github.com/Shinigami92>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+import { EventEmitter } from 'events';
+
+declare namespace Agent {
+	type AgentCallback = (
+		req?: any,
+		opts?: {
+			secureEndpoint: boolean;
+		}
+	) => void;
+
+	interface AgentOptions {
+		timeout?: number;
+		host?: string;
+		port?: number;
+		[key: string]: any;
+	}
+
+	interface Agent extends EventEmitter {
```

Comment:
This says Agent has base as http.Agent
 https://github.com/TooTallNate/node-agent-base#agentfunction-callback-object-options--httpagent

Reply:
Because of
https://github.com/TooTallNate/node-agent-base/blob/master/index.js#L43
I think that `extends http.Agent` is not correct

If I do this, you can call `agent.destory()`, but this method does not exist or is undefined
`typeof agent.destory === undefined`